CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
Well, as I said, I'm not interested in maintaining the use of `SYSTEM`, but I'll take on the work of maintaining suppression of warnings from other headers. I can't take care of this today, but perhaps this weekend I can add what's needed to your PR here and merge it.